.github/workflows/dev-deploy.yaml

```
@@ -0,0 +1,213 @@
+      - uses: actions/checkout@v4
+
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v2
+        with:
+          credentials_json: ${{ secrets.GCP_SA_KEY }}
+
+      - name: Set up gcloud CLI
+        uses: google-github-actions/setup-gcloud@v2
+
+      - name: Deploy Function
+        if: needs.detect_changes.outputs[matrix.function] == 'true'
```

actionlint (1.7.4)

property "detect_changes" is not defined in object type {cloud_function_detect_changes: {outputs: {token_refresh: string}; result: string}}

(expression)